tests/base/src/test/java/org/wildfly/security/auth/client/CredentialStoreSaslAuthenticationTest.java

```diff
@@ -0,0 +1,111 @@
+package org.wildfly.security.auth.client;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.Provider;
+import java.security.Security;
+import java.util.Arrays;
+import java.util.function.Supplier;
+import javax.security.sasl.SaslClient;
+import javax.security.sasl.SaslServer;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wildfly.security.credential.store.CredentialStoreBuilder;
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+import org.wildfly.security.provider.util.ProviderFactory;
+import org.wildfly.security.sasl.plain.PlainSaslServerFactory;
+import org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider;
+import org.wildfly.security.sasl.test.SaslServerBuilder;
+import okhttp3.mockwebserver.MockWebServer;
+
+/**
+ * Tests a successful SASL authentication with a credential store reference in xml configuration
+ *
+ * @author <a href="mailto:szaldana@redhat.com">Sonia Zaldana</a>
+ */
+
+public class CredentialStoreSaslAuthenticationTest {
+
+    private static final String PLAIN = "PLAIN";
+    private static final String USERNAME = "Guest";
+    private static final String PASSWORD = "gpwd";
+    private static final String CREDENTIAL_CONFIG_FILE = "wildfly-credential-sasl-config.xml";
+    private static final MockWebServer server = new MockWebServer();
+    private static String BASE_STORE_DIRECTORY = "target/ks-cred-stores";
+    private static final Supplier<Provider[]> ELYTRON_PROVIDER_SUPPLIER = ProviderFactory.getElytronProviderSupplier(ElytronXmlParser.class.getClassLoader());
+
+    private static final Provider[] providers = new Provider[] {
+            WildFlyElytronSaslPlainProvider.getInstance(),
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server.start(50831);
```

Comment:
I think the `MockWebServer` can be removed since it doesn't look like it's needed for this test.